pylint/reporters/base_reporter.py

```diff
@@ -21,11 +22,10 @@ class BaseReporter:
 
     extension = ""
 
-    def __init__(self, output=None):
-        self.linter = None
+    def __init__(self, output: Optional[TextIO] = None):
```

Comment:
```suggestion
    def __init__(self, output: TextIO = sys.stdout):
```

I think we'll have a better time if we get rid of all the optional ``Optional``.